### PR TITLE
Improve save_as option for visualize()

### LINKED ISSUE #227: Make save-to-file the default behavior for visualize()
Currently the default with no arguments to the IDAES Flowsheet Visualizer is to store the model in memory, but not save at all to disk. It would be better to pick a default place to save and inform the user.

### PR DESCRIPTION
**Pull request by dangunter**
## Summary/Motivation:
- Current behavior may be surprising

## Changes proposed in this PR:
- Add a default save location
- Improve documentation and error handling
-  Return save location to user (modify return type - backwards-incompatible change)

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by makaylas — COMMENTED** [review]
I wish that you separated the doc and psymo changes into a different PR but it's not a big deal

File: idaes/ui/fsvis/fsvis.py [hunk 1]
```diff
@@ -104,7 +171,59 @@ def visualize(
...
+        if not quiet:
+            print("Type ^C to stop the program")
+        while True:
+            time.sleep(1)
+    except KeyboardInterrupt:
+        if not quiet:
+            print("Program stopped")
+
+
+def _pick_default_save_location(name, save_dir):
```
[the remark this answers is not in the document]
Comment (dangunter) [reply idaes/ui/fsvis/fsvis.py:191]: it is only called when a specific save location is not passed in. location != dir

**Review by ksbeattie — APPROVED** [review]
Not being all that familiar with this, LGTM

I also agree that it would be better to put all the style changes in a separate PR

**Review by makaylas — APPROVED** [review]
LGTM

**Pull request merged**